src/fsharp/DotNetFrameworkDependencies.fs

```diff
@@ -34,6 +34,98 @@ module internal FSharp.Compiler.DotNetFrameworkDependencies
                     None
         with _ -> None
 
+    // Compare nuget version strings
+    //
+    // Format:
+    // =======
+    //      $(Major).$(Minor).$(Build) [-SomePrefix]
+    //   Major, Minor, Build collates normally
+    //   Strings without -SomePrefix collate higher than SomePrefix, 
+    //   SomePrefix collates using normal alphanumeric rules
+    //
+    let deconstructVersion (version:string)  =
+        let getSuffix =
+            let pos = version.IndexOf("-")
+            if pos >= 0 then version.Substring(pos + 1) else ""
+        let elements = version.Split('.')
+
+        if elements.Length < 3 then
+            struct (0, 0, 0, getSuffix)
+        else
+            struct (Int32.Parse(elements.[0]), Int32.Parse(elements.[1]), Int32.Parse(elements.[2]), getSuffix)
+
+    let versionCompare c1 c2 =
+        if c1 = c2 then 0
+        else
+            try
+                let struct (major1, minor1, build1, suffix1 ) = deconstructVersion c1
+                let struct (major2, minor2, build2, suffix2 ) = deconstructVersion c2
+
+                let v = major2 - major1
+                if v = 0 then 0
+                else
+                    let v = minor2 - minor1
+                    if v <> 0 then v
+                    else
+                        let v = build2 - build1
+                        if v <> 0 then v
+                        else
+                            String.CompareOrdinal(suffix2, suffix1)
+            with _ -> 0
+
+    let executionTfm =
+        let file =
+            try
+                let depsJsonPath = Path.ChangeExtension(Assembly.GetEntryAssembly().Location, "deps.json")
```

Comment:
This is raising-then-catching exceptions on startup of desktop F# Interactive.  SHould this code be executed for desktop fsi.exe at all?